### PR TITLE
Infer XML namespace using connectHost 

### PR DESCRIPTION
**Pull request by krisis**
While GCS is S3 v4 compatible, it uses a different xml namespace url
than AWS (and Minio).

### CODE REVIEW

File: src/Network/Minio/Data.hs [hunk 1]
```diff
@@ -95,12 +96,38 @@ data ConnectInfo = ConnectInfo {
...
+-- > }
+
+gcsCI :: ConnectInfo
+gcsCI = def {
+      connectHost = "storage.googleapis.com"
+    , connectPort = 443
+    , connectAccessKey = ""
+    , connectSecretKey = ""
+    , connectIsSecure = True
+    , connectSvcNamespace = "http://doc.s3.amazonaws.com/2006-03-01"
```
[the remark this answers is not in the document]
Comment (krisis) [reply src/Network/Minio/Data.hs]: @donatello @harshavardhana so, should I check if `connectHost == "storage.googleapis.com" and use a different xml namespace?

[the remark this answers is not in the document]
Comment (donatello) [reply src/Network/Minio/Data.hs]: Yes, this seems to be a good way.

[the remark this answers is not in the document]
Comment (krisis) [reply src/Network/Minio/Data.hs]: Done

**Comment by donatello** [comment]
There are some buidl warnings:

```
/home/travis/build/minio/minio-hs/src/Network/Minio/JsonParser.hs:30:36: warning: [-Wunused-top-binds]
    Defined but not used: ‘aeCode’
   |
30 | data AdminErrJSON = AdminErrJSON { aeCode    :: Text
   |                                    ^^^^^^
/home/travis/build/minio/minio-hs/src/Network/Minio/JsonParser.hs:31:36: warning: [-Wunused-top-binds]
    Defined but not used: ‘aeMessage’
   |
31 |                                  , aeMessage :: Text
   |                                    ^^^^^^^^^
```

**Comment by harshavardhana** [comment]
> /home/travis/build/minio/minio-hs/src/Network/Minio/JsonParser.hs:30:36: warning: [-Wunused-top-binds]
>     Defined but not used: ‘aeCode’
>    |
> 30 | data AdminErrJSON = AdminErrJSON { aeCode    :: Text
>    |                                    ^^^^^^
> /home/travis/build/minio/minio-hs/src/Network/Minio/JsonParser.hs:31:36: warning: [-Wunused-top-binds]
>     Defined but not used: ‘aeMessage’
>    |
> 31 |                                  , aeMessage :: Text
>    |                                    ^^^^^^^^^

That would be my mistake 

```haskell
data AdminErrJSON = AdminErrJSON { aeCode    :: Text
                                 , aeMessage :: Text
                                 } deriving (Eq, Show)
instance FromJSON AdminErrJSON where
  parseJSON = withObject "AdminErrJSON" $ \v -> AdminErrJSON
    <$> v .: "Code"
    <*> v .: "Message"

parseErrResponseJSON :: (MonadIO m) => LByteString -> m ServiceErr
parseErrResponseJSON jsondata =
  case eitherDecode jsondata of
    Right (AdminErrJSON code message) -> return $ toServiceErr code message
    Left err                          -> throwIO $ MErrVJsonParse $ T.pack err
```

Since we didn't use the aeCode it complains, is there a way to turn this off?

**Pull request merged**